src/shared/previewUtils.ts

```diff
@@ -0,0 +1,430 @@
+  IOSUtils,
+  LaunchArgument,
+  PreviewUtils as LwcDevMobileCorePreviewUtils,
+  Platform,
+} from '@salesforce/lwc-dev-mobile-core';
+import { Progress, Spinner } from '@salesforce/sf-plugins-core';
+import fetch from 'node-fetch';
+
+Messages.importMessagesDirectoryFromMetaUrl(import.meta.url);
+const messages = Messages.loadMessages('@salesforce/plugin-lightning-dev', 'lightning.preview.app');
+const DevPreviewAuraMode = 'DEVPREVIEW';
+
+export class PreviewUtils {
+  public static generateWebSocketUrlForLocalDevServer(platform: string, port: string): string {
+    return LwcDevMobileCorePreviewUtils.generateWebSocketUrlForLocalDevServer(platform, port);
+  }
+
+  /**
+   * Attempts to fetch the targeted mobile device for previewing.
+   *
+   * @param platform A mobile platform (iOS or Android)
+   * @param deviceId An optional device identifier (such as name or UDID)
+   * @param logger An optional logger to be used for logging
+   * @returns The iOS or Android device, or `undefined` if not found
+   */
+  public static async getMobileDevice(
+    platform: Platform.ios | Platform.android,
+    deviceId?: string,
+    logger?: Logger
+  ): Promise<IOSSimulatorDevice | AndroidVirtualDevice | undefined> {
+    let device: IOSSimulatorDevice | AndroidVirtualDevice | undefined;
+
+    logger?.debug(`Attempting to get mobile device for platform ${platform}`);
+
+    if (deviceId) {
+      logger?.debug(`Attempting to get device ${deviceId}`);
+      device =
+        platform === Platform.ios
+          ? (await IOSUtils.getSimulator(deviceId)) ?? undefined
+          : await AndroidUtils.fetchEmulator(deviceId);
+    } else {
+      logger?.debug('No particular device was targeted by the user...  fetching the first available device.');
+      const devices =
+        platform === Platform.ios ? await IOSUtils.getSupportedSimulators() : await AndroidUtils.fetchEmulators();
+      if (devices && devices.length > 0) {
+        device = devices[0];
+      }
+    }
+
+    return Promise.resolve(device);
+  }
+
+  /**
+   * Attempts to boot a device.
+   *
+   * @param platform A mobile platform (iOS or Android)
+   * @param deviceId The identifier (such as name or UDID) of the target device
+   * @param logger An optional logger to be used for logging
+   * @returns For Android devices returns the emulator port number. For iOS devices returns `undefined`
+   */
+  public static async bootMobileDevice(
+    platform: Platform.ios | Platform.android,
+    deviceId: string,
+    logger?: Logger
+  ): Promise<number | undefined> {
+    logger?.debug(`Booting device ${deviceId}`);
+
+    let emulatorPort: number | undefined;
+
+    if (platform === Platform.ios) {
+      await IOSUtils.bootDevice(deviceId, true); // will be no-op if already booted
+      await IOSUtils.launchSimulatorApp();
+    } else {
+      emulatorPort = await AndroidUtils.startEmulator(deviceId); // will be no-op if already booted
+    }
+
+    if (emulatorPort) {
```

Comment:
Can we move the messaging of this conditional into the conditional above? I found myself confused as to where `emulatorPort` would be undefined, only to realize that it's a differentiator of the different platform cases. It'd be more readable just to put these logger messages above.